src/interfaces/ComputeEngine.java

```diff
@@ -0,0 +1,5 @@
+package interfaces;
+
+public interface ComputeEngine {

```

Comment:
if this is the component that's doing the computation then maybe renaming the interface to ComputeEngineComputation would help make it a bit clearer.

Reply:
Sure